src/lib/output/RunResultsProcessor.ts

```diff
@@ -42,7 +55,8 @@ export class RunResultsProcessor {
 			// ...log it to the console...
 			this.display.displayInfo(msg);
 			// ...and return it for use with the --json flag.
-			return msg;
+			this.jsonReturnValueHolder.set(msg);
```

Comment:
It strikes me as weird to have this method update the JSON value holder as a side-effect instead of just returning the results. I understand that it's a consequence of having multiple processors that now run as a group, but still. Problem is I can't think of any alternatives...

Reply:
Yeah, it is a common solution that i've used over the years with the composite pattern. The real thing I want to challenge is why we are even using the '--json' flag in our cli when we have a format=json option. Having a cli based action return a value is rather odd in my opinion.

Reply:
`--json` is a standard flag; I'm not sure we _can_ turn it off. Its format is:
```
{
status: the Exit Code;
result: the AnyJson Returned by the Command's run() method
warnings: array of Warning Messages that would have been logged normally;
}
```
I agree that there's redundancy between it and `--format json` though.

Reply:
Yes we can turn it off. Something for us to consider. It is confusing because there are places in the code where we put things in the display and return stuff in the json return value.